api/proto/teleport/legacy/types/types.proto

```diff
@@ -6422,6 +6424,62 @@ message PluginDatadogAccessSettings {
   string fallback_recipient = 2;
 }
 
+// PluginAWSICSettings holds the settings for an AWS Identity Center integration.
+message PluginAWSICSettings {
+  option (gogoproto.equal) = true;
+
+  // IntegrationName is the Teleport OIDC integration used to gain access to the
+  // AWS account
+  string integration_name = 1;
+
+  // Region is the AWS region the target Identity Center instance is configured in
+  string region = 2;
+
+  // InstanceARN is the arn of the Identity Center instance to manage
+  string arn = 3;
+
+  // Provisioning holds settings for provisioing users and groups into AWS
+  AWSICProvisioningSpec provisioning_spec = 4;
+}
+
+// AWSICProvisioningSpec holds provisioning-specific Identity Center settings
+message AWSICProvisioningSpec {
+  option (gogoproto.equal) = true;
+
+  // BaseURL is the SCIM base URL
+  string base_url = 1;
+
+  // IncludeUsers is a list of predicate expressions used to identify users to
+  // provision into the downstream Identity Center. An empty list implies
+  // "provision all users".
+  repeated string include_users = 2;
+
+  // ExcludeUsers is a list of predicate expressions used to identify users that
+  // must not be provisioned into the downstream Identity Center. ExcludeUsers
+  // overrides IncludeUsers
+  //
+  // An empty list implies "provision all users" (i.e. "exclude no-one")
+  repeated string exclude_users = 3;
+
+  // IncludeAccessLists is a list of predicate expressions used to identify
+  // Access Lists to provision as Groups in the downstream IdP. An empty list
+  // implies "provision all Access Lists".
+  repeated string include_access_lists = 4;
+
+  // ExcludeAccessLists is a list of predicate expressions used to identify
+  // AccessLists that must *not* be provisioned into the downstream Identity
+  // Center. ExcludeAccessLists overrides IncludeAccessLists
+  //
+  // An empty list implies "provision all access lists" (i.e. "exclude no-one")
+  repeated string exclude_access_lists = 5;
+
+  // BearerTokenRef holds a reference to bearer token used to authenticate with
+  // AWS when provisioning users and groups via SCIM. The actual credential can
+  // be looked up in a PluginStaticCredentuials service using the data from
+  // this reference.
+  PluginStaticCredentialsRef bearer_token_ref = 6;
```

Comment:
Isn't this supposed to be the opposite? The spec uses actual `bearer_token` which will be populated with the credential ref during runtime. I think this should be the following no?
```suggestion
  // BearerToken is used to authenticate with AWS when provisioning 
  // users and groups via SCIM. The actual credential is stored as 
  // PluginStaticCredentials reference.
  string bearer_token = 6;
```

Reply:
That's not the way I've done it in other integrations. In my experience, the plugin holds a `PluginStaticCredentialsRef` and the credential is explicitly looked up when needed. That way we can't leak a credential when someone just examines the plugin resource.

Reply:
Ummm iirc that was not the way CredentialsRef was implemented. The plugin (`PluginV1`) defines the `PluginStaticCredentialsRef` but the actual plugin service does not need to know about it.

The `CreatePluginRequest` takes both `PluginV1` and `PluginStaticCredentialsV1`, where `Plugin` holds plugin settings and `StaticCredentials` holds the actual credential. Both are stored separately in the backend. During runtime, plugin manager retrieves static credential for each plugin and injects to the plugin instance factory via `instanceDependencies`. Then finally the plugin factory retrieves its credential from the `instanceDependencies` and configures the actual plugin service. 

For example, picking up this Jamf plugin: 

- See the plugin and static credential config for `CreatePluginRequest` https://github.com/gravitational/teleport.e/blob/master/lib/web/plugindescriptor.go#L281-L318

- During runtime, Plugin manager retrieves static credential for each plugin with `getStaticCredentials` https://github.com/gravitational/teleport.e/blob/master/lib/plugins/manager.go#L387-L402 and adds that credential to the `instanceDependencies` https://github.com/gravitational/teleport.e/blob/master/lib/plugins/manager.go#L308-L326

- And the Jamf plugin retrieves the credential from the `instanceDependencies` https://github.com/gravitational/teleport.e/blob/master/lib/plugins/jamf.go#L15-L34 and uses that credential to configure the Jamf service config https://github.com/gravitational/teleport.e/blob/master/lib/plugins/jamf.go#L39

I checked Okta and it does the same. 
The process is slightly different for the plugins that uses OAuth, where they just store client id and secret to bootstrap which then they exchange for the live token and keep it in memory.

In our case, the `AWSICProvisioningSpec` need not store the ref. Just define a field which stores an empty string in the storage. And during runtime, from the identity center plugin instance factory, we retrieve the credential and then use that to populate the `AWSICProvisioningSpec`.

Reply:
I think I may have gotten confused with how I added the SCIM authentication token in Okta. Will update.

Reply:
Done